Validations/userValidation.js

```diff
@@ -0,0 +1,28 @@
+const Joi = require('joi');
+
+const signupSchema = Joi.object({
+  name: Joi.string().required(),
+  email: Joi.string().email().required(),
+  password: Joi.string().min(6).required(),
+  role: Joi.string().valid('Admin', 'Owner', 'Manager').required(),
+});
+
+const loginSchema = Joi.object({
+  email: Joi.string().email().required(),
+  password: Joi.string().min(6).required(),
+});
+
+const updateUserSchema = Joi.object({
+  name: Joi.string(),
+  email: Joi.string().email(),
+  password: Joi.string().min(6),
+  role: Joi.string().valid('Admin', 'Owner', 'Manager'),
+  isArchive: Joi.boolean(),
+}).or('name', 'email', 'password', 'role', 'isArchive');
```

Comment:
Can you please explain the purpose of this `or` condition?

Reply:
It ensures that at least one of the listed properties should be present in the object being validated. If none of these properties are present, the validation will fail.